client/router.js

```diff
@@ -369,3 +427,45 @@ Router.prototype._updateCallbacks = function () {
 
 Router.prototype._page = page;
 Router.prototype._qs = qs;
+
+function shouldCallTrigger(current, fn) {
+  var name = current.route.name;
+
+  if (typeof fn !== 'function') {
+    return false;
+  }
+
+  if (!name) {
```

Comment:
This one affect to the only logic. So, even if it's only for one route
It's possible to call unnamed routes.